src/v/redpanda/admin_server.cc

```diff
@@ -1748,3 +1753,77 @@ void admin_server::register_hbadger_routes() {
               [] { return ss::json::json_return_type(ss::json::json_void()); });
       });
 }
+
+void admin_server::register_transaction_routes() {
+    ss::httpd::transaction_json::get_all_transactions.set(
+      _server._routes,
+      [this](std::unique_ptr<ss::httpd::request> req)
+        -> ss::future<ss::json::json_return_type> {
+          if (!config::shard_local_cfg().enable_transactions) {
+              throw ss::httpd::bad_request_exception(
+                fmt_with_ctx(fmt::format, "Transaction are disabled"));
+          }
+
+          if (need_redirect_to_leader(model::tx_manager_ntp, _metadata_cache)) {
+              throw co_await redirect_to_leader(*req, model::tx_manager_ntp);
+          }
+
+          auto& tx_frontend = _partition_manager.local().get_tx_frontend();
+          if (!tx_frontend.local_is_initialized()) {
+              throw ss::httpd::bad_request_exception(
+                fmt::format("Can not get tx_frontend"));
+          }
+
+          auto res = co_await tx_frontend.local().get_all_transactions();
+          if (!res.has_value()) {
+              co_await throw_on_error(*req, res.error(), model::tx_manager_ntp);
+          }
+
+          using tx_info = ss::httpd::transaction_json::transaction_summary;
+          std::vector<tx_info> ans;
+          ans.reserve(res.value().size());
+
+          for (auto& tx : res.value()) {
+              if (tx.status == cluster::tm_transaction::tx_status::tombstone) {
+                  continue;
+              }
+
+              tx_info new_tx;
+
+              new_tx.transactional_id = tx.id;
+
+              ss::httpd::transaction_json::producer_identity pid;
+              pid.id = tx.pid.id;
+              pid.epoch = tx.pid.epoch;
+              new_tx.pid = pid;
+
+              new_tx.tx_seq = tx.tx_seq;
+              new_tx.etag = tx.etag;
+              new_tx.status = ss::sstring(tx.get_status());
+              new_tx.timeout_ms = tx.get_timeout().count();
+              new_tx.staleness_ms = tx.get_staleness().count();
+
+              for (auto& partition : tx.partitions) {
+                  ss::httpd::transaction_json::partition partition_info;
+                  partition_info.ns = partition.ntp.ns;
+                  partition_info.topic = partition.ntp.tp.topic;
+                  partition_info.partition_id = partition.ntp.tp.partition;
+                  partition_info.etag = partition.etag;
```

Comment:
can the string copying be avoided in this loop?

Reply:
If I use std::move I got `Passing result of std::move() as a const reference argument; no move will actually happen (fix available)`

Reply:
got it, thanks for the explanation. that's a weakness in Seastar's API. nothing we can do without sending a patch upstream.